src/shared/dissect-image.c

```diff
@@ -393,6 +393,64 @@ static int image_policy_check_protection(
         return 0;
 }
 
+static int partition_is_luks2_integrity(int part_fd, uint64_t offset, uint64_t size, bool use_full_fd, bool *ret) {
```

Comment:
can't we adde the little bit of superblock parsing ourselves to detect integrity stuff? we have a json parser anyway in place. seems pretty straightforward: just read the short  fixed size header off the disk, which is immediately followed by the JSON data. and the json data has the necessary integrity field in there. see lib/luks2/luks2.h in cryptsetup sources, and luks2_json_format.c's LUKS2_generate_hdr(). I think such a minimal parser would not be too hard to get rid. and we have a json parser anyway (sd-json), so it's really short.

Reply:
Implemented!